app/src/main/java/com/github/se/studybuddies/calender/CalendarScreen.kt

```diff
@@ -0,0 +1,189 @@
+package com.github.se.studybuddies.calender
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.KeyboardArrowLeft
+import androidx.compose.material.icons.filled.KeyboardArrowRight
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.se.studybuddies.R
+import com.github.se.studybuddies.data.CalendarUiState
+import com.github.se.studybuddies.navigation.NavigationActions
+import com.github.se.studybuddies.navigation.Route
+import com.github.se.studybuddies.ui.shared_elements.GoBackRouteButton
+import com.github.se.studybuddies.ui.shared_elements.Sub_title
+import com.github.se.studybuddies.ui.shared_elements.TopNavigationBar
+import com.github.se.studybuddies.ui.theme.Blue
+import com.github.se.studybuddies.ui.theme.LightBlue
+import com.github.se.studybuddies.viewModels.CalendarViewModel
+import java.time.YearMonth
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CalendarApp(
+    viewModel: CalendarViewModel = androidx.lifecycle.viewmodel.compose.viewModel(),
+    navigationActions: NavigationActions
+) {
+  val uiState by viewModel.uiState.collectAsState()
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize().testTag("calendar_scaffold"),
+      topBar = {
+        TopNavigationBar(
+            title = { Sub_title(title = "Calendar") },
+            navigationIcon = {
+              GoBackRouteButton(navigationActions = navigationActions, Route.SOLOSTUDYHOME)
+            },
+            actions = {})
+      },
+  ) { padding ->
+    Surface(
+        modifier = Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(padding)) {
+          CalendarWidget(
+              days = DateUtil.daysOfWeek,
+              yearMonth = uiState.yearMonth,
+              dates = uiState.dates,
```

Comment:
Instead of passing yearMonth and dates, you could just give uiState as parameter and take extract yearsMonth and dates inside of the CalendarWidget

Reply:
hey it doesn't change anything , I prefer being explicit on the parameter I need for the function